readthedocs/search/api.py

```diff
@@ -232,6 +232,18 @@ def _get_all_projects_data(self):
                 version_slug=main_version.slug,
                 subproject=project,
             )
+
+            # Fallback to the default version of the subproject.
+            if (
+                not version
+                and main_project.has_feature(Feature.SEARCH_SUBPROJECTS_ON_DEFAULT_VERSION)
+                and project.default_version
```

Comment:
I feel like `project` in this logic should be called `subproject` -- I got a bit confused with this assuming the `project` was the `main_project`.